mteb/abstasks/MTEBResults.py

```diff
@@ -138,9 +142,94 @@ def to_disk(self, path: Path) -> None:
             f.write(self.model_dump_json())
 
     @classmethod
-    def from_disk(cls, path: Path) -> MTEBResults:
+    def from_disk(cls, path: Path, load_historic_data: bool = False) -> MTEBResults:
+        """Load MTEBResults from disk.
+
+        Args:
+            path: The path to the file to load.
+            load_historic_data: Whether to attempt to load historic data from before v1.11.0.
+        """
         with path.open("r") as f:
-            return cls.model_validate(json.load(f))
+            data = json.load(f)
+
+        if not load_historic_data:
+            try:
+                return cls.model_validate(data)
+            except Exception as e:
+                raise ValueError(
+                    f"Error loading MTEBResults from disk. You can try to load historic data by setting `load_historic_data=True`. Error: {e}"
+                )
+
+        err: str = ""
+        try:
+            return cls.model_validate(data)
+        except Exception as e:
+            err = f"Error loading MTEBResults from disk: {e}"
```

Comment:
Rather than raising a ValueError, why don't we just output a warning & directly try to load as historic data if it fails?

Reply:
Yea was considering whether load_historic_data should default be True or False. I can set it to True by default.

Reply:
Set default to True